meteor/server/api/blueprints/context/adlibActions.ts

```diff
@@ -150,6 +153,38 @@ export class ActionExecutionContext extends ShowStyleUserContext implements IAct
 
 		return clone(unprotectObject(lastPieceInstance))
 	}
+
+	findLastScriptedPieceOnLayer(
+		sourceLayerId: string,
+		options?: {
+			excludeCurrentPart?: boolean
+			pieceMetaDataFilter?: any
+		}
+	): IBlueprintPieceDB | undefined {
+		const query: MongoQuery<Piece> = {}
+		if (options && options.pieceMetaDataFilter) {
+			for (const [key, value] of Object.entries(options.pieceMetaDataFilter)) {
+				// TODO do we need better validation here?
+				// It should be pretty safe as we are working with the cache version (for now)
+				query[`piece.metaData.${key}`] = value
+			}
+		}
+
+		if (options && options.excludeCurrentPart && this._cache.Playlist.doc.currentPartInstanceId) {
+			const currentPartInstance = this._cache.PartInstances.findOne(
+				(p) => p._id === this._cache.Playlist.doc.currentPartInstanceId
```

Comment:
You can drop the `(p) => p._id === ` off this query, as the collections do allow finding by giving the id